OpenKeychain/src/main/java/org/sufficientlysecure/keychain/KeychainApplication.java

```diff
@@ -136,20 +136,28 @@ public boolean checkConsolidateRecovery() {
     }
 
     /**
-     * Add OpenKeychain account to Android to link contacts with keys
+     * Add OpenKeychain account to Android to link contacts with keys and keyserver sync
      */
     public static void setupAccountAsNeeded(Context context) {
         try {
             AccountManager manager = AccountManager.get(context);
             Account[] accounts = manager.getAccountsByType(Constants.ACCOUNT_TYPE);
-            if (accounts == null || accounts.length == 0) {
+
+            if (accounts.length == 0) {
                 Account account = new Account(Constants.ACCOUNT_NAME, Constants.ACCOUNT_TYPE);
                 if (manager.addAccountExplicitly(account, null, null)) {
+                    // for contact sync
                     ContentResolver.setIsSyncable(account, ContactsContract.AUTHORITY, 1);
                     ContentResolver.setSyncAutomatically(account, ContactsContract.AUTHORITY, true);
+                    KeyserverSyncAdapterService.enableKeyserverSync(context);
                 } else {
                     Log.e(Constants.TAG, "Adding account failed!");
                 }
+            } else {
+                // users updating to the version with sync will already have the account created
+                if (!Preferences.getPreferences(context).isKeyserverSyncInitialized()) {
+                    KeyserverSyncAdapterService.enableKeyserverSync(context);
+                }
```

Comment:
Instead of using this check and introducing SYNC_KEYSERVER_SETUP, I would prefer that you increase Constants.default.PREF_VERSION and do ``KeyserverSyncAdapterService.enableKeyserverSync(context);`` in Preferences.upgradePreferences.
I know, strictly speaking, enabling the sync adapter is not really a new preference, but because we already have an upgrade mechanism, it fits there.